bundles/io/org.eclipse.smarthome.io.rest.core/src/main/java/org/eclipse/smarthome/io/rest/core/discovery/DiscoveryResource.java

```diff
@@ -91,7 +101,9 @@ public void onFinished() {
             }
         });
 
-        return Response.ok(discoveryServiceRegistry.getMaxScanTimeout(bindingId)).build();
+        DiscoveryServiceDTO dto = new DiscoveryServiceDTO();
+        dto.timeout = discoveryServiceRegistry.getMaxScanTimeout(bindingId);
```

Comment:
Nor really. For a DiscoveryBindingDTO, there should also be a DiscoveryBinding class (for which it is the DTO)...

At this place, isn't all you want to return the timeout integer? What is the point of wrapping this in JSON? Does it really make it much easier for the UIs (and don't we also provide item states as PLAIN)? And if so, is'nt there an easier way to wrap an int into JSON besides defining a whole class (or reusing a different one as you do here) for it?